✨ feat     : attachment, images spec api 작성

CHE1210 opened the pull request:
✨ feat     : attachment, images spec api 작성

attachment serializers, view, url 작성
images serializers, view, url 작성
(파일 업로드(조회) 관련 API)

## ✅ PR 요약
- 관련 이슈 번호: #123
- 작업 요약: 어떤 작업을 했는지 간단하게 적어주세요.

## 📄 상세 내용
- [ ] attachment spec api 작성
- [ ] images spec api 작성
- [ ] aatachment, images url 작성

## 📸 스크린샷 (선택)

## 📝 기타 참고 사항

## 🧪 PR Checklist
- [ ] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [ ] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

CHE1210 committed:
✨ feat     : attachment, images spec api 작성
✨ feat     : attachment, images spec api 작성

attachment serializers, view, url 작성
images serializers, view, url 작성
(파일 업로드(조회) 관련 API)

2joonkim commented on apps/recruitments/serializers/attachment.py:
```diff
@@ -0,0 +1,17 @@
+    updated_at = serializers.DateTimeField(read_only=True)
+
+
+class Meta:
```
이거 들여쓰기가 좀 이상한데요 채희님..? 전체적으로 들여쓰기가 좀 이상한거같은데... 확인 한번 해주시죠!!

2joonkim commented on apps/recruitments/serializers/attachment.py:
```diff
@@ -0,0 +1,17 @@
+from apps.recruitments.models.attachment import RecruitmentAttachment
+
+
+class AttachmentSerializer(serializers.ModelSerializer[RecruitmentAttachment]):
```
이 부분도 좀 이상합니다..!!

2joonkim commented on apps/recruitments/serializers/attachment.py:
```diff
@@ -0,0 +1,17 @@
+    file_name = serializers.CharField(max_length=50)
+    created_at = serializers.DateTimeField(read_only=True)
+    updated_at = serializers.DateTimeField(read_only=True)
+
```
지금 그리고 ModelSerializer를 쓰셨는데 커스텀이 필요한 항목들에 대해서만
기입을 해주시면 될 거 같습니다! 기본적으로 들어있는 내용이 중복이 될 수 있습니다 수정해주세요

2joonkim commented on apps/recruitments/serializers/images.py:
```diff
@@ -0,0 +1,15 @@
+from apps.recruitments.models.recruitment_images import RecruitmentImage
+
+
+class ImagesSerializer(serializers.ModelSerializer[RecruitmentImage]):
```
이 부분도 위 코멘트와 동일합니다

2joonkim commented on apps/recruitments/views/attachment.py:
```diff
@@ -0,0 +1,131 @@
+
+        # recruitment 유효성 (Mock)
+        if recruitment_id in self.MOCK_NONEXISTENT_RECRUITMENT_IDS:
+            return Response({"detail": "존재하지 않는 공고입니다."}, status=status.HTTP_400_BAD_REQUEST)
```
여긴 400보단 404가 더 맞을 거 같습니다 한번 확인해보세요!!

2joonkim commented on apps/recruitments/views/attachment.py:
```diff
@@ -0,0 +1,131 @@
+    )
+    def delete(self, request: Request, attachment_id: int, *args: Any, **kwargs: Any) -> Response:
+        # 팀 코드 스타일에 맞춰 204와 메시지를 함께 반환
+        return Response({"detail": "첨부파일이 삭제되었습니다."}, status=status.HTTP_204_NO_CONTENT)
```
여기도 204보단 200이 맞을 거 같습니다
첨부파일이 삭제되었단 메세지를 노출 시키기 위해 (확인 부탁드리겠습니다)

2joonkim commented on apps/recruitments/views/images.py:
```diff
@@ -0,0 +1,120 @@
+import uuid
```
이 부분도 에러코드 확인 한번만 부탁드리겠습니다!

CHE1210 committed:
💡 chore    : attachment, images serializers 코드 수정 및 attachment view 에러 확인 코드 수정
💡 chore    : attachment, images serializers 코드 수정 및 attachment view 에러 확인 코드 수정

attachment, images serializers ModelSerializer를 변경, 들여쓰기 수정, view 에러 코드 부분 변경
